custom_components/adaptive_lighting/switch.py

```diff
@@ -1316,7 +1316,9 @@ async def _update_manual_control_and_maybe_adapt(
                 else:
                     _fire_manual_control_event(self, light, context)
             else:
-                await self._adapt_light(light, transition, force=force, context=context)
+                self.hass.async_create_task(
```

Comment:
This whole PR is now these 3 lines change. 

Why does this make it adapt the lights simultaneously?

Reply:
The 3-line changes the behavior from adapting each light in the loop* and `await`ing one by one, to *starting adapt tasks as parallel threads* one by one.

The `await` adds enough delay to be noticeable, and more so if the lights don't respond fast enough:
```
>> start adapting light1 -> await response -> [response + communication delay] -> done with light1 -> start adapting light2 -> etc.
```

Making it parallel minimizes that delay as much as possible by not waiting for each light to finish the `_adapt_light` task:
```
>> start adapting light1 -> [response + communication delay] -> done with light1
>> start adapting light2 -> [response + communication delay] -> done with light2
etc.
```


*the loop being the line `for light in filtered_lights:`


(Came across this issue while researching about that small delay I was experiencing with multiple lights)

Reply:
Well said, to add to this the change is most visible in scenarios where the adapt_delay is nonzero